chain/actors/actor_storagemarket.go

```diff
@@ -122,15 +122,21 @@ func (sdp *StorageDealProposal) Cid() (cid.Cid, error) {
 	return nd.Cid(), nil
 }
 
-func (sdp *StorageDealProposal) Verify() error {
-	unsigned := *sdp
-	unsigned.ProposerSignature = nil
-	var buf bytes.Buffer
-	if err := unsigned.MarshalCBOR(&buf); err != nil {
-		return err
+func (sdp *StorageDealProposal) Verify(worker address.Address) error {
+	if sdp.Client != worker || worker == address.Undef {
```

Comment:
Can worker ever be `undef` in this case, we verify that `worker` is same as `vmctx.Message().From`

Reply:
Yeah, so this is there because we use the Verify message in other parts of the codebase (when accepting a deal over the network) where we *always* want to check the signature

Reply:
Ahh, I didn't notice that.
It is weird calling to part of actor's code for that but it makes some sense.